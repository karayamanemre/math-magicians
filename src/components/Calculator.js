import React from 'react';
import calculate from '../logic/calculate.js';

class Calculator extends React.Component {
  constructor(props) {
    super(props);
    this.state = {
      total: 0,
      next: null,
      operation: null,
    };
    this.onClickHandler = this.onClickHandler.bind(this);
  }
  handleClick(e) {
    this.setState((state) => calculate(state, e.target.innerText));
  }
  onClickHandler(e) {
    this.setState((state) => calculate(state, e.target.innerText));
  }
  render() {
    const { next, total, operation } = this.state;
    return (
      <div className="container">
        <span className="display">
          {total}
          {operation}
          {next}
        </span>
        <button className="clear" onClick={this.onClickHandler}>
          AC
        </button>
        <button className="plus-minus" onClick={this.onClickHandler}>
          +/-
        </button>
        <button className="percent" onClick={this.onClickHandler}>
          %
        </button>
        <button className="operator divide" onClick={this.onClickHandler}>
          รท
        </button>
        <button className="number" onClick={this.onClickHandler}>
          7
        </button>
        <button className="number" onClick={this.onClickHandler}>
          8
        </button>
        <button className="number" onClick={this.onClickHandler}>
          9
        </button>
        <button className="operator multiply" onClick={this.onClickHandler}>
          x
        </button>
        <button className="number" onClick={this.onClickHandler}>
          4
        </button>
        <button className="number" onClick={this.onClickHandler}>
          5
        </button>
        <button className="number" onClick={this.onClickHandler}>
          6
        </button>
        <button className="operator minus" onClick={this.onClickHandler}>
          -
        </button>
        <button className="number" onClick={this.onClickHandler}>
          1
        </button>
        <button className="number" onClick={this.onClickHandler}>
          2
        </button>
        <button className="number" onClick={this.onClickHandler}>
          3
        </button>
        <button className="operator plus" onClick={this.onClickHandler}>
          +
        </button>
        <button className="number zero" onClick={this.onClickHandler}>
          0
        </button>
        <button className="decimal" onClick={this.onClickHandler}>
          .
        </button>
        <button className="operator equal" onClick={this.onClickHandler}>
          =
        </button>
      </div>
    );
  }
}

export default Calculator;
