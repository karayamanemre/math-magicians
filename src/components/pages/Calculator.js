import React, { useState } from 'react';
import calculate from '../../logic/calculate.js';

const Calculator = () => {
  const [state, setState] = useState({ total: 0, next: null, operation: null });
  const onClickHandler = (e) => {
    setState(() => calculate(state, e.target.innerText));
  };
  const { total, operation, next } = state;
  return (
    <section className="calculator">
      <h2>Let's Do Some Math!</h2>
      <div className="calculatorContainer">
        <span className="display">
          {total}
          {operation}
          {next}
        </span>
        <button className="clear" onClick={onClickHandler}>
          AC
        </button>
        <button className="plus-minus" onClick={onClickHandler}>
          +/-
        </button>
        <button className="percent" onClick={onClickHandler}>
          %
        </button>
        <button className="operator divide" onClick={onClickHandler}>
          รท
        </button>
        <button className="number" onClick={onClickHandler}>
          7
        </button>
        <button className="number" onClick={onClickHandler}>
          8
        </button>
        <button className="number" onClick={onClickHandler}>
          9
        </button>
        <button className="operator multiply" onClick={onClickHandler}>
          x
        </button>
        <button className="number" onClick={onClickHandler}>
          4
        </button>
        <button className="number" onClick={onClickHandler}>
          5
        </button>
        <button className="number" onClick={onClickHandler}>
          6
        </button>
        <button className="operator minus" onClick={onClickHandler}>
          -
        </button>
        <button className="number" onClick={onClickHandler}>
          1
        </button>
        <button className="number" onClick={onClickHandler}>
          2
        </button>
        <button className="number" onClick={onClickHandler}>
          3
        </button>
        <button className="operator plus" onClick={onClickHandler}>
          +
        </button>
        <button className="number zero" onClick={onClickHandler}>
          0
        </button>
        <button className="decimal" onClick={onClickHandler}>
          .
        </button>
        <button className="operator equal" onClick={onClickHandler}>
          =
        </button>
      </div>
    </section>
  );
};

export default Calculator;
