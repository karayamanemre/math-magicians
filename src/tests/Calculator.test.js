import { render } from '@testing-library/react';
import Calculator from '../components/pages/Calculator';

describe('test for creating', () => {
  it('renders correctly', () => {
    const view = render(<Calculator />);
    expect(view).toMatchSnapshot();
  });
});
