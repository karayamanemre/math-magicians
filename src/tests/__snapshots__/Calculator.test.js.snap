// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test rendering and user interaction renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="calculator"
      >
        <h2>
          Let's Do Some Math!
        </h2>
        <div
          class="calculatorContainer"
        >
          <span
            class="display"
          >
            0
          </span>
          <button
            class="clear"
          >
            AC
          </button>
          <button
            class="plus-minus"
          >
            +/-
          </button>
          <button
            class="percent"
          >
            %
          </button>
          <button
            class="operator divide"
          >
            รท
          </button>
          <button
            class="number"
          >
            7
          </button>
          <button
            class="number"
          >
            8
          </button>
          <button
            class="number"
          >
            9
          </button>
          <button
            class="operator multiply"
          >
            x
          </button>
          <button
            class="number"
          >
            4
          </button>
          <button
            class="number"
          >
            5
          </button>
          <button
            class="number"
          >
            6
          </button>
          <button
            class="operator minus"
          >
            -
          </button>
          <button
            class="number"
          >
            1
          </button>
          <button
            class="number"
          >
            2
          </button>
          <button
            class="number"
          >
            3
          </button>
          <button
            class="operator plus"
          >
            +
          </button>
          <button
            class="number zero"
          >
            0
          </button>
          <button
            class="decimal"
          >
            .
          </button>
          <button
            class="operator equal"
          >
            =
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="calculator"
    >
      <h2>
        Let's Do Some Math!
      </h2>
      <div
        class="calculatorContainer"
      >
        <span
          class="display"
        >
          0
        </span>
        <button
          class="clear"
        >
          AC
        </button>
        <button
          class="plus-minus"
        >
          +/-
        </button>
        <button
          class="percent"
        >
          %
        </button>
        <button
          class="operator divide"
        >
          รท
        </button>
        <button
          class="number"
        >
          7
        </button>
        <button
          class="number"
        >
          8
        </button>
        <button
          class="number"
        >
          9
        </button>
        <button
          class="operator multiply"
        >
          x
        </button>
        <button
          class="number"
        >
          4
        </button>
        <button
          class="number"
        >
          5
        </button>
        <button
          class="number"
        >
          6
        </button>
        <button
          class="operator minus"
        >
          -
        </button>
        <button
          class="number"
        >
          1
        </button>
        <button
          class="number"
        >
          2
        </button>
        <button
          class="number"
        >
          3
        </button>
        <button
          class="operator plus"
        >
          +
        </button>
        <button
          class="number zero"
        >
          0
        </button>
        <button
          class="decimal"
        >
          .
        </button>
        <button
          class="operator equal"
        >
          =
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;

exports[`test for creating renders correctly 1`] = `
Object {
  "asFragment": [Function],
  "baseElement": <body>
    <div>
      <section
        class="calculator"
      >
        <h2>
          Let's Do Some Math!
        </h2>
        <div
          class="calculatorContainer"
        >
          <span
            class="display"
          >
            0
          </span>
          <button
            class="clear"
          >
            AC
          </button>
          <button
            class="plus-minus"
          >
            +/-
          </button>
          <button
            class="percent"
          >
            %
          </button>
          <button
            class="operator divide"
          >
            รท
          </button>
          <button
            class="number"
          >
            7
          </button>
          <button
            class="number"
          >
            8
          </button>
          <button
            class="number"
          >
            9
          </button>
          <button
            class="operator multiply"
          >
            x
          </button>
          <button
            class="number"
          >
            4
          </button>
          <button
            class="number"
          >
            5
          </button>
          <button
            class="number"
          >
            6
          </button>
          <button
            class="operator minus"
          >
            -
          </button>
          <button
            class="number"
          >
            1
          </button>
          <button
            class="number"
          >
            2
          </button>
          <button
            class="number"
          >
            3
          </button>
          <button
            class="operator plus"
          >
            +
          </button>
          <button
            class="number zero"
          >
            0
          </button>
          <button
            class="decimal"
          >
            .
          </button>
          <button
            class="operator equal"
          >
            =
          </button>
        </div>
      </section>
    </div>
  </body>,
  "container": <div>
    <section
      class="calculator"
    >
      <h2>
        Let's Do Some Math!
      </h2>
      <div
        class="calculatorContainer"
      >
        <span
          class="display"
        >
          0
        </span>
        <button
          class="clear"
        >
          AC
        </button>
        <button
          class="plus-minus"
        >
          +/-
        </button>
        <button
          class="percent"
        >
          %
        </button>
        <button
          class="operator divide"
        >
          รท
        </button>
        <button
          class="number"
        >
          7
        </button>
        <button
          class="number"
        >
          8
        </button>
        <button
          class="number"
        >
          9
        </button>
        <button
          class="operator multiply"
        >
          x
        </button>
        <button
          class="number"
        >
          4
        </button>
        <button
          class="number"
        >
          5
        </button>
        <button
          class="number"
        >
          6
        </button>
        <button
          class="operator minus"
        >
          -
        </button>
        <button
          class="number"
        >
          1
        </button>
        <button
          class="number"
        >
          2
        </button>
        <button
          class="number"
        >
          3
        </button>
        <button
          class="operator plus"
        >
          +
        </button>
        <button
          class="number zero"
        >
          0
        </button>
        <button
          class="decimal"
        >
          .
        </button>
        <button
          class="operator equal"
        >
          =
        </button>
      </div>
    </section>
  </div>,
  "debug": [Function],
  "findAllByAltText": [Function],
  "findAllByDisplayValue": [Function],
  "findAllByLabelText": [Function],
  "findAllByPlaceholderText": [Function],
  "findAllByRole": [Function],
  "findAllByTestId": [Function],
  "findAllByText": [Function],
  "findAllByTitle": [Function],
  "findByAltText": [Function],
  "findByDisplayValue": [Function],
  "findByLabelText": [Function],
  "findByPlaceholderText": [Function],
  "findByRole": [Function],
  "findByTestId": [Function],
  "findByText": [Function],
  "findByTitle": [Function],
  "getAllByAltText": [Function],
  "getAllByDisplayValue": [Function],
  "getAllByLabelText": [Function],
  "getAllByPlaceholderText": [Function],
  "getAllByRole": [Function],
  "getAllByTestId": [Function],
  "getAllByText": [Function],
  "getAllByTitle": [Function],
  "getByAltText": [Function],
  "getByDisplayValue": [Function],
  "getByLabelText": [Function],
  "getByPlaceholderText": [Function],
  "getByRole": [Function],
  "getByTestId": [Function],
  "getByText": [Function],
  "getByTitle": [Function],
  "queryAllByAltText": [Function],
  "queryAllByDisplayValue": [Function],
  "queryAllByLabelText": [Function],
  "queryAllByPlaceholderText": [Function],
  "queryAllByRole": [Function],
  "queryAllByTestId": [Function],
  "queryAllByText": [Function],
  "queryAllByTitle": [Function],
  "queryByAltText": [Function],
  "queryByDisplayValue": [Function],
  "queryByLabelText": [Function],
  "queryByPlaceholderText": [Function],
  "queryByRole": [Function],
  "queryByTestId": [Function],
  "queryByText": [Function],
  "queryByTitle": [Function],
  "rerender": [Function],
  "unmount": [Function],
}
`;
